microstructure_ve.py

```diff
@@ -304,19 +304,51 @@ def load_matlab_microstructure(matfile, var_name):
     return loadmat(matfile, matlab_compatible=True)[var_name]
 
 
+# def assign_intph(microstructure, num_layers):
+#     """Generate interphase layers around the particles.
+
+#     Particles must be falsey, matrix must be truthy
+#     """
+#     from scipy.ndimage import distance_transform_edt
+
+#     dists = distance_transform_edt(microstructure)
+#     particles: np.ndarray = dists == 0
+#     matrix = dists > num_layers
+#     interphase = ~(matrix | particles)
+#     return 2 * matrix.view("u1") + interphase.view("u1")
+
 def assign_intph(microstructure, num_layers):
     """Generate interphase layers around the particles.
-
     Particles must be falsey, matrix must be truthy
     """
     from scipy.ndimage import distance_transform_edt
 
-    dists = distance_transform_edt(microstructure)
+    # special case, matrix with 1 element
+    if len(np.unique(microstructure))==1:
+        return microstructure
+    # assign 0 to all non-matrix positions, assuming matrix positions have the max value
+    matrix_only = (microstructure==microstructure.max())
+    dists = distance_transform_edt(matrix_only)
     particles: np.ndarray = dists == 0
     matrix = dists > num_layers
-    interphase = ~(matrix | particles)
-    return 2 * matrix.view("u1") + interphase.view("u1")
-
+#     interphase = ~(matrix | particles)
+    return microstructure + matrix.view("u1")
+
+def assign_multi_intph(microstructure, num_layers_list):
```

Comment:
Rather than create a second function, I think it would be nicer to augment the original function to take a list. It is easy enough to make a list with one item if desired.